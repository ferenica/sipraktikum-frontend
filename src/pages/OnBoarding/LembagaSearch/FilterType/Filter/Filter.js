import React from 'react';

import Checkbox from '@material-ui/core/Checkbox';
import {styled} from '@material-ui/core/styles';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import PropTypes from 'prop-types';

const StyledFormControlLabel = styled(FormControlLabel)({
  label: {
    fontSize: '14px',
  },
});

const Filter = (props) => {
  return (
    <StyledFormControlLabel
      control={
        <Checkbox
          data-testid={props.value.nama}
          name={props.value.nama}
          onChange={props.checkboxHandler}
          checked={props.value.isChecked}
        />}
      label={props.value.nama}
      key={props.index}
    />
  );
};

export default Filter;

Filter.propTypes = {
  value: PropTypes.object,
  checkboxHandler: PropTypes.func,
  index: PropTypes.number,
};
