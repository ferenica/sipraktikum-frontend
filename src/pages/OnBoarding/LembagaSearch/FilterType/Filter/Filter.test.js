import React from 'react';
import Filter from './Filter';
import Enzyme, {shallow, mount, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('Filter', () => {
  it('is implemented correctly', () => {
    const props = {
      value: {
        nama: 'Praktikum 1',
        isChecked: false,
      },
      label: 'Praktikum 1',
    };
    const wrapper = mount(<Filter {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
