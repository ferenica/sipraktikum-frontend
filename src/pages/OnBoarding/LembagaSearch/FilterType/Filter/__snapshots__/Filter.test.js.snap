// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter is implemented correctly 1`] = `ReactWrapper {}`;
